src/Process/Runner/Parallel/ParallelProcessRunner.php

```diff
@@ -120,22 +145,38 @@ private function cleanFinished(): bool
         return false;
     }
 
-    private function startProcess(MutantProcess $mutantProcess): bool
+    private function startProcess(ProcessBearer $processBearer): bool
     {
-        $mutant = $mutantProcess->getMutant();
+        $processBearer->getProcess()->start();
 
-        if (!$mutant->isCoveredByTest()) {
-            $this->eventDispatcher->dispatch(new MutantProcessWasFinished(
-                MutantExecutionResult::createFromProcess($mutantProcess)
-            ));
+        $this->currentProcesses[] = $processBearer;
 
-            return false;
-        }
+        return true;
+    }
 
-        $mutantProcess->getProcess()->start();
+    /**
+     * @param ProcessBearer[] $bucket
+     * @param Generator|ProcessBearer[] $input
```

Comment:
Won't work for my IDE :(

Reply:
yes but at least it's correct, otherwise it's interpreted as `Generator<mixed>|ProcessBearer[]` which is... not good. But it works sometimes for me so I wonder if the PHP extensions plugin has anything to do with this

Reply:
What interprets it as such? I recall `Iterator|Object[]` was invented and/or popularised by PHPStorm folks, so virtually every other tool can read it as `Iterator<Object>`. More importantly I think PHPStan is fine with this.

Reply:
>What interprets it as such?

phpStorm, or maybe it's inferring from somewhere else but that's what I get quite often in those cases :/

It's a convention pushed by both PHPStan and psalm and IMO make a lot more sense and that we adopted so far elsewhere in the codebase

Reply:
We can see that [PHPStan is fine with iterables](https://phpstan.org/r/4d335ef9-f576-430a-82ba-30e10b744fd4), but other Iterators are lacking such understanding. I'm going to use the syntax you propose for them as you propose.